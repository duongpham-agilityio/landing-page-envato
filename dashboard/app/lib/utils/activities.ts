import { AxiosError } from 'axios';

// Store
import { authStore } from '../stores';

// Format
import { formatUppercaseFirstLetter } from '.';

// Interface
import { TActivitiesRequest } from '../interfaces';

// Service
import { MainHttpService, recentActivitiesHttpService } from '../services';

// Constants
import { END_POINTS } from '../constants';

export const handleActivities = async (url: string, actionName: string) => {
  const { user } = authStore();
  try {
    const { data } = await MainHttpService.axiosClient.get(url);

    if (data && user) {
      await recentActivitiesHttpService.post<TActivitiesRequest>(
        END_POINTS.RECENT_ACTIVITIES,
        {
          userId: user.id,
          actionName: actionName,
        },
      );
    }
  } catch (error) {
    const { response } = error as AxiosError<string>;

    throw new Error(formatUppercaseFirstLetter(response?.data));
  }
};
