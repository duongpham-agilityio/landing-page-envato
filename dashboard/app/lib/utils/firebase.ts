import { initializeApp } from 'firebase/app';
import { getAuth } from 'firebase/auth';
import { getStorage } from 'firebase/storage';
import { getFirestore } from 'firebase/firestore';
import { Messaging, getToken } from 'firebase/messaging';

const firebaseConfig = {
  apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
  authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
  projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
  storageBucket: process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET,
  messagingSenderId: process.env.NEXT_PUBLIC_FIREBASE_SENDER_ID,
  appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID,
  measurementId: process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID,
};

initializeApp(firebaseConfig);

const VAPID_KEY = process.env.NEXT_PUBLIC_FIREBASE_VAPID;

export const requestForToken = (messaging: Messaging) =>
  getToken(messaging, { vapidKey: VAPID_KEY })
    .then((currentToken) => {
      if (currentToken) {
        console.log('current token for client: ', currentToken);

        return currentToken;
        // Perform any other neccessary action with the token
      } else {
        // Show permission request UI
        console.log(
          'No registration token available. Request permission to generate one.',
        );
      }
    })
    .catch((err) => {
      console.log('An error occurred while retrieving token. ', err);
    });

// Initialize Firebase
export const app = initializeApp(firebaseConfig);
export const auth = getAuth();
export const storage = getStorage();
export const db = getFirestore();
