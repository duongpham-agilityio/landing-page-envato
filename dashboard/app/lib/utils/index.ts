export * from './transaction';
export * from './time';
export * from './user';
export * from './url';
export * from './helpers';
export * from './text';
export * from './localStorage';
export * from './image';
export * from './prefetch';
export * from './toast';
