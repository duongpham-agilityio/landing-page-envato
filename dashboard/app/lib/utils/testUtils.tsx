import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactNode } from 'react';

const queryClient = new QueryClient();

export const renderQueryProviderTest = (component: JSX.Element) => {
  const { render } = testLibReactUtils;
  return render(
    <QueryClientProvider client={queryClient}>{component}</QueryClientProvider>,
  );
};

export const queryProviderWrapper = ({ children }: { children: ReactNode }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);
