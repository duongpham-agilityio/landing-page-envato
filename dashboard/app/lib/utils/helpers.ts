// Constants
import { REGEX, DOTS, ERROR_MESSAGES } from '@/lib/constants';

// Types
import { FormatPaginationParams, PaginationTableType } from '@/lib/interfaces';

export const formatNumberButton = (numberOfPage: number): string[] =>
  Array.from({ length: numberOfPage }).map((_, index: number): string =>
    (index + 1).toString(),
  );

export const formatPageArray = ({
  totalPage = 0,
  currentPage = 0,
}: PaginationTableType): string[] => {
  // If the number of pages is less than or equal to 5, display all page buttons
  if (totalPage <= 5) {
    return formatNumberButton(totalPage);
  }

  // If the current page is near the start (pages 1 to 3),
  // Display the first 3 pages and then DOTS and the last page
  if (currentPage <= 3) {
    return [...formatNumberButton(3), DOTS, totalPage.toString()];
  }

  // If the current page is near the end (within the last 3 pages),
  // display the first page, DOTS, and the last 3 pages
  if (currentPage >= totalPage - 2) {
    return ['1', DOTS, ...formatNumberButton(totalPage).splice(-3)];
  }

  // If the current page is in the middle,
  // Display the first page, DOTS, currentPage, DOTS, and the last page
  return ['1', DOTS, currentPage.toString(), DOTS, totalPage.toString()];
};

export const formatPagination = ({
  totalCount,
  limit,
  currentPage,
}: FormatPaginationParams): string[] => {
  const totalPage = Math.ceil(totalCount / limit);

  return formatPageArray({ totalPage, currentPage });
};

export const validatePassword = (value: string) => {
  if (!value) {
    return ERROR_MESSAGES.FIELD_REQUIRED('Password');
  }

  if (!REGEX.LENGTH_IS_EIGHT.test(value)) {
    return ERROR_MESSAGES.PASS_WORD_SHORT;
  }

  if (!REGEX.PASSWORD.test(value)) {
    return ERROR_MESSAGES.PASS_WORD_WEAK;
  }

  return true;
};

export const formatUppercaseFirstLetter = (value = ''): string =>
  value.charAt(0).toUpperCase() + value.slice(1);

export const formatDecimalInput = (value = ''): string => {
  const validData = /(^[0-9])[^.]*((?:\.\d*)?)/.exec(
    value.replace(/[^(\d|.)]/g, ''),
  );

  const formatValue = validData ? validData[0] : '';

  return formatValue;
};

export const formatAmountNumber = (value: string): string => {
  if (!value) {
    return value;
  }

  if (Number.isNaN(parseFloat(value))) {
    return '';
  }

  const dotIndex = value.indexOf('.');
  const decimalValue = dotIndex > -1 ? value.slice(dotIndex) : '';
  const newValue =
    dotIndex > -1
      ? value.slice(0, dotIndex).replaceAll(',', '')
      : value.replaceAll(',', '');

  const newValueFormat = newValue.replace(/\B(?=(\d{3})+(?!\d))/g, ',');

  if (
    !REGEX.AMOUNT_PATTERN.test(newValue) ||
    !REGEX.DECIMAL_PATTERN.test(decimalValue)
  ) {
    return `${newValueFormat}.${decimalValue
      .substring(0, 3)
      .replaceAll(/[.]/g, '')}`;
  }

  return decimalValue ? `${newValueFormat}${decimalValue}` : newValueFormat;
};

/**
 * Remove amount format ex: 12,345.00 -> 12345
 * @param amount string
 * @returns number
 */
export const removeAmountFormat = (amount: string): number =>
  +amount.replaceAll(',', '');

export const parseFormattedNumber = (value: string): number => {
  if (!value) {
    return 0;
  }

  const cleanValue = String(value).replace(REGEX.FORMAT_NUMBER, '');

  const parsedNumber = parseFloat(cleanValue);

  if (Number.isNaN(parsedNumber)) {
    return 0;
  }

  return parsedNumber;
};

/**
 * Format number rg: 12345 -> 12,345.00 if isOmitDecimals = false or 12,345 if isOmitDecimals = true
 * @param number
 * @param isOmitDecimals
 * @returns Number after format
 */
export const formatDecimalNumber = (
  number = 0,
  isOmitDecimals: boolean = false,
): string => {
  const formattedNumber = isOmitDecimals
    ? Math.round(number).toString()
    : number.toFixed(2);

  const numberWithCommas = formattedNumber.replace(
    /\B(?=(\d{3})+(?!\d))/g,
    ',',
  );

  return numberWithCommas;
};

// Format for typing numbers in input
export const formatAllowOnlyNumbers = (input: string): string => {
  if (!input) {
    return ''; // or handle the case when input is undefined or null
  }

  // Replace any non-digit character with an empty string
  return input.replace(/[^0-9]/g, '');
};

export const getStatusColor = (status: string): string => {
  switch (status.toLowerCase()) {
    case 'online':
      return 'green.500';
    case 'offline':
      return 'gray.500';
    default:
      return 'gray.500';
  }
};

export const isWindowDefined = () => typeof window !== 'undefined';
