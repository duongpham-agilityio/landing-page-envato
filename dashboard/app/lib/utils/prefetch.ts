// Libs
import { QueryClient } from '@tanstack/react-query';

// Services
import { MainHttpService } from '@/lib/services';

export const prefetchStatistical = async <T>(
  endPoint: string,
  queryClient: QueryClient,
) => {
  await queryClient.prefetchQuery({
    queryKey: [endPoint],
    queryFn: async () =>
      (await MainHttpService.get<T>({ path: endPoint })).data,
  });
};
