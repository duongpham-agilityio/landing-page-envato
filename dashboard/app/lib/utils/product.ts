import dayjs from 'dayjs';

// Types
import { Status, TProduct } from '@/lib/interfaces';

// Utils
import { formatDecimalNumber, formatUppercaseFirstLetter } from '.';

// Constants
import { IMAGES, TIME_FORMAT } from '../constants';

/**
 * Convert data show for home page
 * @param products
 * @returns
 */
export const formatProductResponse = (products: TProduct[] = []) =>
  products.map((product) => {
    const {
      _id,
      name,
      amount,
      currency,
      createdAt,
      description,
      imageURLs,
      stock,
    } = product;

    return {
      id: _id,
      name: formatUppercaseFirstLetter(name),
      description: formatUppercaseFirstLetter(description),
      date: dayjs(createdAt).format(TIME_FORMAT),
      imageURLs: imageURLs || IMAGES.BIG_AVATAR.url,
      amount: `${currency}${formatDecimalNumber(+amount)}`,
      stock: `${stock}`,
      status: stock > 0 ? Status.IN_STOCK : Status.SOLD,
    };
  });
