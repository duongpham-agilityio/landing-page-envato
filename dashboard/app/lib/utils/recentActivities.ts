import dayjs from 'dayjs';

// Types
import { TRecentActivities } from '@/lib/interfaces';

// Utils
import { formatUppercaseFirstLetter } from '.';

// Constants
import { TIME_FORMAT } from '../constants';

/**
 * Convert data show for home page
 * @param recentActivities
 * @returns
 */
export const formatRecentActivitiesResponse = (
  recentActivities: TRecentActivities[] = [],
) =>
  recentActivities.map((activity) => {
    const { _id, actionName, email, createdAt } = activity;

    return {
      _id: _id,
      name: formatUppercaseFirstLetter(actionName),
      email,
      date: dayjs(createdAt).format(TIME_FORMAT),
    };
  });
