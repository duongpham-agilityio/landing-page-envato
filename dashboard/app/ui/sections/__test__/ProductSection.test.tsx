import preloadAll from 'jest-next-dynamic';
import { ReactElement, ReactNode } from 'react';
import { useRouter, useSearchParams } from 'next/navigation';

// Utils
import { renderQueryProviderTest } from '@/lib/utils/testUtils';

// Sections
import { ProductsSection } from '..';

jest.mock('react-intersection-observer');

jest.mock('react-intersection-observer', () => ({
  InView: ({
    children,
  }: {
    children: (props: { inView: boolean; ref: () => void }) => ReactNode;
  }) => children({ inView: true, ref: jest.fn() }) as ReactElement,
}));

describe('ProductSection render', () => {
  beforeEach(async () => {
    await preloadAll();
    (useRouter as jest.Mock).mockImplementation(() => ({
      push: jest.fn(),
    }));
    (useSearchParams as jest.Mock).mockImplementation(
      () => new URLSearchParams(),
    );

    (useSearchParams as jest.Mock).mockReturnValue({
      get: jest.fn(),
    });
  });

  test('Should render match with snapshot.', () => {
    const { container } = renderQueryProviderTest(<ProductsSection />);

    expect(container).toMatchSnapshot();
  });
});
