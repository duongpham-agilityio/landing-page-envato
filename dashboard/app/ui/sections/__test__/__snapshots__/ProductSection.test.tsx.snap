// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSection render Should render match with snapshot. 1`] = `
<div>
  <div
    class="css-1qfa3yr"
  >
    <div
      class="css-rklm6r"
    >
      <section
        class="css-1gg9e9z"
      >
        <div
          class="css-194coc3"
        >
          <form
            class="chakra-stack css-16wym0o"
            data-testid="search-bar"
          >
            <div
              class="css-1b9rej7"
            >
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <div
                  class="chakra-input__group css-1091msq"
                  data-group="true"
                >
                  <div
                    class="chakra-input__left-element css-1kg931p"
                  >
                    <svg
                      class="chakra-icon css-onkibi"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M23.414,20.591l-4.645-4.645a10.256,10.256,0,1,0-2.828,2.829l4.645,4.644a2.025,2.025,0,0,0,2.828,0A2,2,0,0,0,23.414,20.591ZM10.25,3.005A7.25,7.25,0,1,1,3,10.255,7.258,7.258,0,0,1,10.25,3.005Z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  <input
                    class="chakra-input css-1x84oq"
                    data-testid="search-transaction"
                    id="field-:r0:"
                    placeholder="Search by name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="css-tzs3vb"
            >
              <button
                aria-controls="menu-list-:r2:"
                aria-expanded="false"
                aria-haspopup="menu"
                class="chakra-button chakra-menu__menu-button css-10e6i5m"
                id="menu-button-:r2:"
                type="button"
              >
                <span
                  class="css-xl71ch"
                >
                  <span
                    class="css-ccj5l2"
                  >
                    <svg
                      fill="none"
                      height="17"
                      stroke="#1D1E24"
                      width="18"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M7.552 13.502H1.25M10.362 3.81h6.301"
                      />
                      <path
                        clip-rule="evenodd"
                        d="M5.948 3.756a2.355 2.355 0 0 0-2.363-2.347 2.355 2.355 0 0 0-2.363 2.347 2.355 2.355 0 0 0 2.363 2.346 2.355 2.355 0 0 0 2.363-2.346ZM17.221 13.463a2.354 2.354 0 0 0-2.362-2.346 2.355 2.355 0 0 0-2.364 2.346 2.355 2.355 0 0 0 2.364 2.347 2.354 2.354 0 0 0 2.362-2.347Z"
                      />
                    </svg>
                    Filters
                  </span>
                </span>
              </button>
              <div
                class="css-r6z5ec"
                style="visibility: hidden; position: absolute; inset: 0 auto auto 0;"
              >
                <div
                  aria-orientation="vertical"
                  class="chakra-menu__menu-list css-1k4isgz"
                  id="menu-list-:r2:"
                  role="menu"
                  style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
                  tabindex="-1"
                >
                  <button
                    class="chakra-menu__menuitem css-o9weuj"
                    data-index="0"
                    id="menu-list-:r2:-menuitem-:r3:"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  >
                    <p
                      class="chakra-text css-15asmb0"
                    >
                      Default
                    </p>
                  </button>
                  <button
                    class="chakra-menu__menuitem css-o9weuj"
                    data-index="1"
                    id="menu-list-:r2:-menuitem-:r4:"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  >
                    <p
                      class="chakra-text css-15asmb0"
                    >
                      Sold
                    </p>
                  </button>
                  <button
                    class="chakra-menu__menuitem css-o9weuj"
                    data-index="2"
                    id="menu-list-:r2:-menuitem-:r5:"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  >
                    <p
                      class="chakra-text css-15asmb0"
                    >
                      In stock
                    </p>
                  </button>
                </div>
              </div>
            </div>
          </form>
          <button
            aria-label="Add User"
            class="chakra-button css-gpwyvs"
            data-testid="button-add"
            role="button"
            type="button"
          >
            Add Product
          </button>
        </div>
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="chakra-skeleton css-s55j40"
        />
        <div
          class="css-cyp3zu"
          data-testid="pagination"
        >
          <div
            class="css-70qvj9"
          >
            <p
              class="chakra-text css-r5mmfv"
            >
              Show result:
            </p>
            <div
              class="css-110fpu2"
            >
              <button
                aria-controls="menu-list-:r7:"
                aria-expanded="false"
                aria-haspopup="menu"
                class="chakra-button chakra-menu__menu-button css-10e6i5m"
                id="menu-button-:r7:"
                type="button"
              >
                <span
                  class="css-xl71ch"
                >
                  <div
                    class="css-1n2mv2k"
                  >
                    <span
                      class="chakra-text css-1ftaikx"
                    >
                      10
                    </span>
                    <div
                      class="css-zaxsyz"
                    >
                      <svg
                        fill="none"
                        height="15"
                        stroke="#A0AEC0"
                        style="transform: rotate(undefined);"
                        viewBox="0 0 24 24"
                        width="18"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7 10L12 14L17 10"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                        />
                      </svg>
                    </div>
                  </div>
                </span>
              </button>
              <div
                class="css-r6z5ec"
                style="visibility: hidden; position: absolute; inset: 0 auto auto 0;"
              >
                <div
                  aria-orientation="vertical"
                  class="chakra-menu__menu-list css-1k4isgz"
                  id="menu-list-:r7:"
                  role="menu"
                  style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
                  tabindex="-1"
                >
                  <button
                    class="chakra-menu__menuitem css-o9weuj"
                    data-index="0"
                    id="menu-list-:r7:-menuitem-:r8:"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  >
                    <p
                      class="chakra-text css-15asmb0"
                    >
                      10
                    </p>
                  </button>
                  <button
                    class="chakra-menu__menuitem css-o9weuj"
                    data-index="1"
                    id="menu-list-:r7:-menuitem-:r9:"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  >
                    <p
                      class="chakra-text css-15asmb0"
                    >
                      20
                    </p>
                  </button>
                  <button
                    class="chakra-menu__menuitem css-o9weuj"
                    data-index="2"
                    id="menu-list-:r7:-menuitem-:ra:"
                    role="menuitem"
                    tabindex="-1"
                    type="button"
                  >
                    <p
                      class="chakra-text css-15asmb0"
                    >
                      50
                    </p>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-69i1ev"
          >
            <button
              aria-label="btn-prev"
              class="chakra-button css-1vy1nhu"
              data-testid="prev-button"
              disabled=""
              type="button"
            >
              <svg
                fill="none"
                height="24"
                stroke="#A0AEC0"
                style="transform: rotate(90deg);"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 10L12 14L17 10"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
            <div
              class="css-70qvj9"
            />
            <button
              aria-label="btn-next"
              class="chakra-button css-5dzvs"
              data-testid="next-button"
              type="button"
            >
              <svg
                fill="none"
                height="24"
                stroke="#A0AEC0"
                style="transform: rotate(-90deg);"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 10L12 14L17 10"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
      </section>
    </div>
    <div
      class="css-1bm8fk6"
    >
      <div
        class="css-1n88nru"
      >
        <div
          class="css-1p0sw78"
        >
          <h3
            class="chakra-heading css-1jhdndu"
          >
            my wallet
          </h3>
          <div
            class="css-gmuwbf"
          >
            <div
              class="css-140b0vm"
            >
              <div
                class="css-1loy0ln"
              >
                <p
                  class="chakra-text css-mskrhv"
                >
                  Balance
                </p>
                <button
                  aria-label="eye"
                  class="chakra-button css-he8p55"
                  data-testid="btn-eye"
                  type="button"
                >
                  <svg
                    height="16"
                    viewBox="0 0 576 512"
                    width="18"
                  >
                    <g
                      opacity="0.6"
                    >
                      <path
                        d="M288 80c-65.2 0-118.8 29.6-159.9 67.7C89.6 183.5 63 226 49.4 256c13.6 30 40.2 72.5 78.6 108.3C169.2 402.4 222.8 432 288 432s118.8-29.6 159.9-67.7C486.4 328.5 513 286 526.6 256c-13.6-30-40.2-72.5-78.6-108.3C406.8 109.6 353.2 80 288 80zM95.4 112.6C142.5 68.8 207.2 32 288 32s145.5 36.8 192.6 80.6c46.8 43.5 78.1 95.4 93 131.1c3.3 7.9 3.3 16.7 0 24.6c-14.9 35.7-46.2 87.7-93 131.1C433.5 443.2 368.8 480 288 480s-145.5-36.8-192.6-80.6C48.6 356 17.3 304 2.5 268.3c-3.3-7.9-3.3-16.7 0-24.6C17.3 208 48.6 156 95.4 112.6zM288 336c44.2 0 80-35.8 80-80s-35.8-80-80-80c-.7 0-1.3 0-2 0c1.3 5.1 2 10.5 2 16c0 35.3-28.7 64-64 64c-5.5 0-10.9-.7-16-2c0 .7 0 1.3 0 2c0 44.2 35.8 80 80 80zm0-208a128 128 0 1 1 0 256 128 128 0 1 1 0-256z"
                        fill="white"
                        opacity="1"
                      />
                    </g>
                  </svg>
                </button>
              </div>
              <p
                class="chakra-text css-llqfbq"
              >
                ******
              </p>
            </div>
          </div>
          <form
            class="css-1ayfwcb"
          >
            <p
              class="chakra-text css-y05pe7"
            >
              quick transfer
            </p>
            <div
              class="css-79elbk"
            >
              <div
                class="chakra-input__group css-1lq0mp5"
                data-group="true"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <div
                    class="chakra-input__group css-1091msq"
                    data-group="true"
                  >
                    <input
                      class="chakra-input css-18ql1xp"
                      id="field-:ri:"
                      name="memberId"
                      placeholder="Choose an account to transfer"
                      type="text"
                      value=""
                    />
                    <button
                      aria-label="The eye icon"
                      class="chakra-input__right-element css-1stj06r"
                      data-testid="right-icon-input"
                      type="button"
                    >
                      <button
                        aria-label="Icon dropdown"
                        class="chakra-button css-1rb8ldf"
                        type="button"
                      >
                        <svg
                          fill="none"
                          height="21"
                          style="transform: translateY(4px);"
                          viewBox="0 0 21 21"
                          width="21"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M7.72168 5.03271L12.7217 10.0327L17.7217 5.03271"
                            stroke="#A0AEC0"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                          />
                        </svg>
                      </button>
                    </button>
                  </div>
                </div>
                <div
                  class="chakra-input__right-element css-f46tp1"
                >
                  <button
                    aria-label="Icon dropdown"
                    class="chakra-button css-1rb8ldf"
                    type="button"
                  >
                    <svg
                      fill="none"
                      height="21"
                      style="transform: translateY(4px);"
                      viewBox="0 0 21 21"
                      width="21"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M7.72168 5.03271L12.7217 10.0327L17.7217 5.03271"
                        stroke="#A0AEC0"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="css-1hlzswh"
            >
              <p
                class="chakra-text css-1ki0mgu"
              >
                 Enter amount 
              </p>
              <div
                class="css-17piex8"
              >
                <p
                  class="chakra-text css-4xj0bk"
                >
                  $
                </p>
                <div
                  class="chakra-form-control css-1eor3tg"
                  role="group"
                >
                  <input
                    autocomplete="off"
                    class="chakra-input css-cz5fpu"
                    id="field-:rj:"
                    name="amount"
                    placeholder="0.00"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <button
              aria-label="btn-send-money"
              class="chakra-button css-1mu22hy"
              disabled=""
              type="submit"
            >
              Send Money
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
